File: 2017/02/1.js
"use strict";

const stdin = process.stdin;

stdin.resume();
stdin.setEncoding('utf8');

stdin.on('data', function (data) {
  data = data.replace(/^\s+|\s+$/, '');

  const sum = data.split("\n").reduce((sum, line) => {
    const row = line.split(/\s+/).map((n) => parseInt(n, 10));
    const max = row.reduce((max, n) => Math.max(max, n));
    const min = row.reduce((min, n) => Math.min(min, n));
    const checksum = max - min;

    return sum + checksum;
  }, 0);

  console.log(sum);
});

File: 2015/02/1.rb
sum = $stdin.reduce(0) do |sum, line|
  l, w, h = line.chomp.split('x').map(&:to_i)
  products = [l*w, w*h, h*l]
  sum + products.map {|n| 2*n }.reduce(:+) + products.min
end

puts sum
